core/src/wallet/ripple/synchronizers/RippleLikeAccountSynchronizer.hpp

```diff
@@ -0,0 +1,145 @@
+/*
+ *
+ * RippleLikeAccountSynchronizer
+ *
+ * Created by El Khalil Bellakrid on 06/01/2019.
+ *
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2019 Ledger
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *
+ */
+
+
+#ifndef LEDGER_CORE_RIPPLELIKEACCOUNTSYNCHRONIZER_H
+#define LEDGER_CORE_RIPPLELIKEACCOUNTSYNCHRONIZER_H
+
+#include <wallet/ripple/keychains/RippleLikeKeychain.h>
+#include <wallet/ripple/explorers/RippleLikeBlockchainExplorer.h>
+#include <wallet/pool/WalletPool.hpp>
+#include <async/DedicatedContext.hpp>
+#include <events/ProgressNotifier.h>
+
+#include <memory>
+#include <mutex>
+
+namespace ledger {
+namespace core {
+
+class RippleLikeAccount;
+
```

Comment:
line 49- line 82
I find the code was copied from AbstractBlockchainExplorerAccountSynchronizer.h
because they are same structs. I think you can use the ones in AbstractBlockchainExplorerAccountSynchronizer.h
instead of copy-paste directly here. 
We can discuss offline if you want